### ISSUE TITLE
Download Plugin Data removes achievement progress on Steam

### ISSUE THREAD

**Issue by Whitechapel-1888** [Lacro59/playnite-successstory-plugin#444]
Success Story 3.1.7

I updated my entire game library using "Main Menu > Extensions > Success Story > Download plugin data...". This downloaded the correct achievements for all games fixing [this issue](https://github.com/Lacro59/playnite-successstory-plugin/issues/442), but removed all progress from my steam games:
![image](https://github.com/Lacro59/playnite-successstory-plugin/assets/44060684/8f2f1cd5-2a31-4f68-a904-46ac87560db4)

I completed this game some time ago:
![image](https://github.com/Lacro59/playnite-successstory-plugin/assets/44060684/447ca2fb-96ce-4193-b341-cb60080b5e00)

RetroAchievements & GoG are not affected.

**Comment by Lacro59** [comment]
Can you post your extension.log?

**Comment by Whitechapel-1888** [comment]
> Can you post your extension.log?

I would but there's no error notification, so no log dump is created. Is the extension.log accessible somewhere else?

**Comment by damirarh** [comment]
The same thing is happening to me in the last two versions. There are no errors, here are the relevant entries from the log for the problematic scan:

```

08-01 18:31:34.870|INFO |SuccessStory#PluginDatabaseObject`4:RefreshNoLoader(Half-Life: Alyx - e413dbda-4685-4eac-9697-bb2afd950834)
08-01 18:31:34.870|INFO |SuccessStory#PluginDatabaseObject`4:Used SuccessStory.Clients.SteamAchievements for Half-Life: Alyx - e413dbda-4685-4eac-9697-bb2afd950834
08-01 18:31:34.870|INFO |SuccessStory#SteamAchievements:GetAchievements() - IsLocal : False, IsManual : False, HasApiKey: False, SteamIsPrivate: False
08-01 18:31:35.306|INFO |SuccessStory#SteamAchievements:SteamApi.GetAchievements()
08-01 18:31:37.284|INFO |SuccessStory#PluginDatabaseObject`4:0/42 achievements find for Half-Life: Alyx - e413dbda-4685-4eac-9697-bb2afd950834
```

I haven't had time to debug the issue yet. Maybe something fails to parse because of a different locale.

**Comment by damirarh** [comment]
Ok, I did some debugging. As far as I can tell, the issue is caused by this `if` statement:

https://github.com/Lacro59/playnite-successstory-plugin/blob/032c98d45b9ce12f950c091263eb7f8e17d7b910/source/Clients/SteamAchievements.cs#L90

At this point `steamAchievements` contains all the achievements but without any unlock data because I didn't enter a Steam API key. Because of this my `accountInfos` is `null` and the whole block below is skipped (not marking any of my achievements as unlocked):

https://github.com/Lacro59/playnite-plugincommon/blob/95b985867adc9ae54b043ad3d13a4be733637474/CommonPluginsStores/Steam/SteamApi.cs#L424-L456

If I change the `if` statement above to force the code to continue in the `else` block, then it correctly scrapes the unlocked achievements. I changed as follows:

```csharp
var accountInfos = SteamApi.CurrentAccountInfos;
ObservableCollection<GameAchievement> steamAchievements = SteamApi.GetAchievements(game.GameId, accountInfos);
if (steamAchievements?.Count > 0 && accountInfos != null && accountInfos.IsCurrent)
```

I didn't create a PR because I'm not sure that this change won't break something else.

**Comment by OdinIsLord** [comment]
Same, all steam games are at 0% progress.

**Comment by PatrykJWisniewski** [comment]
I was having this issue as well. Never had to use an API key before, but adding one fixed this issue for me.

**Comment by Whitechapel-1888** [comment]
> I was having this issue as well. Never had to use an API key before, but adding one fixed this issue for me.

I have added an API key but the steam achievements still do not show up. I updated the game library, downloaded the SuccessStory plugin data, and refreshed individually and nothing worked.

**Comment by OdinIsLord** [comment]
> > I was having this issue as well. Never had to use an API key before, but adding one fixed this issue for me.
> 
> I have added an API key but the steam achievements still do not show up. I updated the game library, downloaded the SuccessStory plugin data, and refreshed individually and nothing worked.

Yeah I kinda gave up on it. Funny how the only games left that don't have achievements working are the games I paid for.

**Comment by damirarh** [comment]
> > I was having this issue as well. Never had to use an API key before, but adding one fixed this issue for me.
> 
> I have added an API key but the steam achievements still do not show up. I updated the game library, downloaded the SuccessStory plugin data, and refreshed individually and nothing worked.

@Whitechapel-1888 Did you add the API key to SuccessStory settings (they are separate from the Steam integrations settings)? To add the API key for SuccessStory, navigate to Main Menu > Add-ons... and then Extension settings > Generic > SuccessStory. Click the Libraries tab, expand Steam and enter the API Key there.
![image](https://github.com/Lacro59/playnite-successstory-plugin/assets/1292060/eca17878-7c9f-4e3c-84fe-9d9d160859ad)

**Comment by Whitechapel-1888** [comment]
@damirarh I wasn't aware that there was an extra field to enter the API key. 

So, is there a specific reason why SuccessStory needs this extra field and cannot authenticate via the API key saved in playnite? Also, the fact that this works surprises me. From the code snippet above, I guessed the issue would apply to everyone eventually, regardless of the method of authentification.

**Comment by DarylGraves** [comment]
I've had a similar situation today. I'm not 100% sure how I fixed it as it takes ages to scan each time I make a change but I think unchecking "Always keep my total playtime private even if users can see my game details." may have been the issue for me.

I did also add the API key in as well but I have tried adding and removing that multiple times today and didn't have luck on it's own.

These are my Steam settings in the end:

![image](https://github.com/Lacro59/playnite-successstory-plugin/assets/46203429/92959bc1-a4da-42cb-94b7-e81ab0622cf9)
